docs/user-guides/plugin/plugin-tests.md

```diff
@@ -0,0 +1,139 @@
+data:
+  run.sh: |-
+
+    #!/usr/bin/env bats
+
+    load "/usr/lib/bats/bats-detik/utils"
+    load "/usr/lib/bats/bats-detik/detik"
+
+    DETIK_CLIENT_NAME="kubectl"
+
+    @test "Verify successful deployment and running status of the {{ .Release.Name }}-operator pod" {
+        verify "there is 1 deployment named '{{ .Release.Name }}-operator'"
+        verify "there is 1 service named '{{ .Release.Name }}-operator'"
+        try "at most 2 times every 5s to get pods named '{{ .Release.Name }}-operator' and verify that '.status.phase' is 'running'"
+    }
+
+    @test "Verify successful creation and bound status of {{ .Release.Name }} persistent volume claims" {
+        try "at most 3 times every 5s to get persistentvolumeclaims named '{{ .Release.Name }}.*' and verify that '.status.phase' is 'Bound'"
+    }
+
+    @test "Verify successful creation and available replicas of {{ .Release.Name }} Prometheus resource" {
+        try "at most 3 times every 5s to get prometheuses named '{{ .Release.Name }}' and verify that '.status.availableReplicas' is more than '0'"
+    }
+
+    @test "Verify creation of required custom resource definitions (CRDs) for {{ .Release.Name }}" {
+        verify "there is 1 customresourcedefinition named 'prometheuses'"
+        verify "there is 1 customresourcedefinition named 'podmonitors'"
+    }
+{{- end -}}
+```
+
+> **Note:** You can use [this guide](https://github.com/bats-core/bats-detik/blob/master/examples/bats/test_kubectl_and_oc.sh) for reference when writing your test assertions.
+
+3. **Test Pod Definition**:
+
+   - Create a `test-<plugin-name>.yaml` file in the `templates/tests` directory to define a `Pod` that will run your tests.
+   - This test `Pod` will mount the `ConfigMap` created in the previous step and will execute the test script `run.sh`.
+
+```yaml
+ {{- if .Values.testFramework.enabled -}}
+ apiVersion: v1
+ kind: Pod
+ metadata:
+   name: {{ .Release.Name }}-test
+   namespace: {{ .Release.Namespace }}
+   labels:
+     type: integration-test
+   annotations:
+     "helm.sh/hook": test
+     "helm.sh/hook-delete-policy": "before-hook-creation,hook-succeeded"
+ spec:
+   serviceAccountName: {{ .Release.Name }}-test
+   containers:
+     - name: bats-test
+       image: "{{ .Values.testFramework.image.registry}}/{{ .Values.testFramework.image.repository}}:{{ .Values.testFramework.image.tag }}"
+       imagePullPolicy: {{ .Values.testFramework.image.pullPolicy }}
+       command: ["bats", "-t", "/tests/run.sh"]
+       volumeMounts:
+         - name: tests
+           mountPath: /tests
+           readOnly: true
+   volumes:
+     - name: tests
+       configMap:
+         name: {{ .Release.Name }}-test
+   restartPolicy: Never
+ {{- end -}}
+```
+
+4. **RBAC Permissions**:
+
+- Create a `test-permissions.yaml` file in the `templates/tests` folder to define the `ServiceAccount` and necessary RBAC permissions for the test Pod.
```

Comment:
```suggestion
- Create the necessary RBAC resources in the `templates/tests` folder with a dedicated `ServiceAccount` and role authorisations so that the test `Pod` can cover test the cases. 
```